utils/to_string.hh

```diff
@@ -63,25 +61,10 @@ std::ostream& operator<<(std::ostream& os, const print_with_comma<NeedsComma, Pr
 
 namespace std {
 
-template<typename Printable>
-static inline
-sstring
-to_string(const std::vector<Printable>& items) {
-    return "[" + utils::join(", ", items) + "]";
-}
-
-template<typename Printable>
-static inline
-sstring
-to_string(const std::set<Printable>& items) {
-    return "{" + utils::join(", ", items) + "}";
-}
-
-template<typename Printable>
-static inline
+template <std::ranges::range Range>
 sstring
-to_string(const std::unordered_set<Printable>& items) {
-    return "{" + utils::join(", ", items) + "}";
+to_string(const Range& items) {
+    return fmt::format("{{{}}}", fmt::join(items, ", "));
 }
```

Comment:
Should we even have to_string(range)? Isn't it better for each call site to decide on its own how to print it?

Reply:
Note that within a formatting snippet, fmt::join() is more efficient since it never allocates an intermediate string, but using to_string() does.

Reply:
Agreed.  those to_string definitions are gone in the next couple patches.
I'll add a FIXME in the next version so it would be more evident.
I just didn't want to do both in a single patch, which is complex enough as it is.